spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AbstractAmqpOutboundEndpoint.java

```diff
@@ -193,10 +194,22 @@ public void setLazyConnect(boolean lazyConnect) {
 	}
 
 	/**
+	 * Set the value to set in the {@code x-delay} header when using the
+	 * RabbitMQ delayed message exchange plugin. By default, the {@link AmqpHeaders#DELAY}
+	 * header (if present) is mapped; setting the delay here overrides that value.
+	 * @param delayExpression the expression.
+	 * @since 5.0
+	 */
+	public void setDelay(int delay) {
+		this.delayExpression = new SupplierExpression<>(() -> delay);
```

Comment:
Good observation about `delay-expression="12"`, but that is only a tip of the iceberg.
The SpEL parser produces `SpelExpression` anyway. It isn't so heavy though with just `IntLiteral` in the `ast`, but still.
I don't say that we should go with additional `delay` attribute, but keep in mind a SpEL nature would be good idea anyway.

Merging...